python/pyarrow/_flight.pyx

```diff
@@ -1866,41 +1919,49 @@ cdef CStatus _do_put(void* self, const CServerCallContext& context,
         FlightDescriptor descriptor = \
             FlightDescriptor.__new__(FlightDescriptor)
 
+    server_state = <object> c_state
     descriptor.descriptor = reader.get().descriptor()
     py_reader.reader.reset(reader.release())
     py_writer.writer.reset(writer.release())
     try:
-        (<object> self).do_put(ServerCallContext.wrap(context), descriptor,
-                               py_reader, py_writer)
+        server_state.this.do_put(ServerCallContext.wrap(context), descriptor,
+                                 py_reader, py_writer)
         return CStatus_OK()
```

Comment:
Nit: The general convention seems to be to place `return CStatus_OK()` at end of function.